### ISSUE TITLE
Fused optimizer

### ISSUE THREAD

**Issue by florianscheidl** [ecmwf/WeatherGenerator#2543]
### Describe the task. It can be a feature, documentation, etc.

The fused implementation of Adam can be slightly faster (see https://docs.pytorch.org/docs/2.9/generated/torch.optim.Adam.html).

For the config_operan_georing_avhrr_forecasting_lowres config, we've seen a small speedup in the optimizer step, see https://gitlab.jsc.fz-juelich.de/hedgedoc/GupBj3vJTHqyW1VM_jKK5Q?both#Removing-gradient-scaling.

### Area

- [ ] datasets, data readers, data preparation and transfer
- [ ] model
- [ ] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation
- [x] performance

**Comment by florianscheidl** [comment]
Assign me
